### ISSUE TITLE
Development Server Error

### ISSUE THREAD

**Issue by mpresland** [tomwalder/php-gds#151]
Hi Tom,

I haven't touched any of my GAE projects for quite a while, and recently updated the SDK and development environment for GAE, and ever since can't use the development server. I keep getting a fatal error: "Uncaught exception 'google\appengine\runtime\ApplicationError' with message 'Illegal string &quot;dev~None&quot; in dataset id."

After updating gcloud I was forced to remove the Application ID and version from my .yaml and wonder if this is causing issues with the development environment? I recall you assisted me in setting the ApplicationID using the $_SERVER array for unit testing, but this doesn't seem to bring any joy for this issue.

Do you have any advice on how to resolve this? My current alternative is to continually deploy to GAE, which makes development incredibly slow... 

Any help would be great.

Cheers,

Matt

**Comment by mpresland** [comment]
`Fatal error: Uncaught exception 'google\appengine\runtime\ApplicationError' with message 'Illegal string "dev~None" in dataset id.' in /Users/mpresland/google-cloud-sdk/platform/google_appengine/php/sdk/google/appengine/runtime/RealApiProxy.php:53 Stack trace: #0 /Users/mpresland/google-cloud-sdk/platform/google_appengine/php/sdk/google/appengine/runtime/ApiProxy.php(40): google\appengine\runtime\RealApiProxy->makeSyncCall('datastore_v4', 'Lookup', Object(google\appengine\datastore\v4\LookupRequest), Object(google\appengine\datastore\v4\LookupResponse), 60) #1 /Users/mpresland/Dropbox/Projects/Web/fitb/fitbbookingportal/vendor/tomwalder/php-gds/src/GDS/Gateway/ProtoBuf.php(220): google\appengine\runtime\ApiProxy::makeSyncCall('datastore_v4', 'Lookup', Object(google\appengine\datastore\v4\LookupRequest), Object(google\appengine\datastore\v4\LookupResponse), 60) #2 /Users/mpresland/Dropbox/Projects/Web/fitb/fitbbookingportal/vendor/tomwalder/php-gds/src/GDS/Gateway/ProtoBuf.php(253): GDS\Gateway\ProtoBuf->execute('Lookup', in /Users/mpresland/google-cloud-sdk/platform/google_appengine/php/sdk/google/appengine/runtime/RealApiProxy.php on line 53`

**Comment by tomwalder** [comment]
Hey there.

Sorry for the delayed reply. I am run off my feet at the moment.

I've not had the issue myself, but I expect it's related to the tilde (~)

Are you using the latest version of GDS?

**Comment by mpresland** [comment]
Hi Tom,

This only happened after updating gcloud, each time I tried to deploy a message instructed me to remove the
application: 
version:
fields from my app.yaml file before deploying to App Engine. Once I did this I could not longer use the dev server, as there was no application id set. I found a workaround though, simply created a dev.yaml that includes the application: field, and left it out of the production app.yaml file. When using the dev server, I start the server using dev.yaml and it seems to be working just fine.
